Adyen.Test/PayoutTest.cs

```diff
@@ -33,11 +36,11 @@ public class PayoutTest : BaseTest
         [TestMethod]
         public void StoreDetailAndSubmitThirdPartySuccessTest()
         {
-            var client = CreateMockTestClientNullRequiredFieldsRequest("Mocks/payout/storeDetailAndSubmitThirdParty-success.json");
-            var payout = new Payout(client);
+            var client = CreateMockTestClientApiKeyBasedRequestAsync("Mocks/payout/storeDetailAndSubmitThirdParty-success.json");
```

Comment:
We're using the client creation method here for `CreateMockTestClientApiKeyBasedRequestAsync`, but the payout.StoreDEtailsAndSubmitPayout is calling the synchronous method, I assume that this test passes because we indirectly call the asynchronous call (since it indirectly calls the async version). Could you verify whether the test succeeds/fails? 

The main difference is in its setup, see snippet in BaseTest.cs:
```csharp
/// diff. between CreateMockTestClientNullRequiredFieldsRequest and CreateMockTestClientApiKeyBasedRequestAsync

/// CreateMockTestClientApiKeyBasedRequestAsync
   var confMock = MockPaymentData.CreateConfigApiKeyBasedMock(); // diff 1
    ClientInterfaceMock.Setup(x => x.RequestAsync(It.IsAny<string>(),
        It.IsAny<string>(), It.IsAny<RequestOptions>(), It.IsAny<HttpMethod>())).ReturnsAsync(response); 
   var confMock = MockPaymentData.CreateConfigApiKeyBasedMock(); // diff 2
              
/// CreateMockTestClientNullRequiredFieldsRequest 
      clientInterfaceMock // diff 1
          .Setup(x => x.Request(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<RequestOptions>(), null))
          .Returns(response);      
      clientInterfaceMock // diff 1
          .Setup(x => x.RequestAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<RequestOptions>(), null))
          .Returns(Task.FromResult(response));              
      var confMock = MockPaymentData.CreateConfigMock(); // diff 2
```

Reply:
Yes indeed. Since every method now is pretty much an async method being awaited I have to use async mock tests for them as well. They all pass after converting them to this mock client.